presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaMetadata.java

```diff
@@ -189,11 +203,52 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     @Override
     public List<ConnectorTableLayoutResult> getTableLayouts(ConnectorSession session, ConnectorTableHandle table, Constraint<ColumnHandle> constraint, Optional<Set<ColumnHandle>> desiredColumns)
     {
-        InformationSchemaTableHandle handle = (InformationSchemaTableHandle) table;
-        ConnectorTableLayout layout = new ConnectorTableLayout(new InformationSchemaTableLayoutHandle(handle, constraint.getSummary()));
+        InformationSchemaTableHandle handle = checkTableHandle(table);
+
+        TupleDomain<ColumnHandle> predicate = constraint.getSummary();
+        Optional<Stream<Map<ColumnHandle, NullableValue>>> partitions = getPartitions(session, handle.getSchemaTableName(), handle.getCatalogName(), predicate);
+        if (partitions.isPresent()) {
+            List<TupleDomain<ColumnHandle>> partitionsTupleDomains = partitions.get()
+                    .filter(constraint.predicate())
```

Comment:
constraint can also have `column_name` right?

Reply:
Possibly. However, it would be too fine-grained to consider `column_name` here.